bundles/org.openhab.binding.noolite/src/main/java/org/openhab/binding/noolite/handler/NooliteMTRF64BridgeHandler.java

```diff
@@ -0,0 +1,206 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.noolite.handler;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.noolite.NooLiteBindingConstants;
+import org.openhab.binding.noolite.internal.NooliteMTRF64Adapter;
+import org.openhab.binding.noolite.internal.config.NooliteBridgeConfiguration;
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author Petr Shatsillo - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class NooliteMTRF64BridgeHandler extends BaseBridgeHandler {
+
+    private static Logger logger = LoggerFactory.getLogger(NooliteMTRF64BridgeHandler.class);
+    @Nullable
+    static NooliteMTRF64Adapter adapter;
+    @Nullable
+    private NooliteBridgeConfiguration bridgeConfig;
+    public static Map<String, NooliteHandler> thingHandlerMap = new HashMap<String, NooliteHandler>();
+    @Nullable
+    static NooliteHandler nooliteHandler;
+
+    public NooliteMTRF64BridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Dispose Noolite bridge handler {}", this.toString());
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "Close bridge");
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Noolite bridge handler {}", this.toString());
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "Initializing...");
+        bridgeConfig = getConfigAs(NooliteBridgeConfiguration.class);
+
+        // logger.debug("Checking Noolite adapter connection. {}", thing.getStatus());
+        if (thing.getStatus() != ThingStatus.ONLINE) {
+            connect();
+        }
+    }
+
+    @SuppressWarnings("null")
+    private void connect() {
+        logger.debug("Connecting to Noolite adapter");
+        try {
+            if (bridgeConfig.serial != null) {
+                if (adapter == null) {
+                    adapter = new NooliteMTRF64Adapter();
```

Comment:
This `adapter` looks like it can just be created as a final field here in the class. So you don't have do the null checks.